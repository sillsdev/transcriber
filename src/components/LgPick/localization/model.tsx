// WARNING: This file is generated using ToModel.xsl. Changes made here may be lost.
import * as Localize from 'react-localization';

export interface ILanguagePickerStrings extends Localize.LocalizedStringsMethods {
    "font": string;
    "script": string;
    "language": string;
    "selectLanguage": string;
    "findALanguage": string;
    "codeExplained": string;
    "subtags": string;
    "details": string;
    "languageOf": string;
    "inScript": string;
    "select": string;
    "cancel": string;
};


export interface ILocalizedStrings {
	loaded: boolean;
	lang: string;
	languagePicker: ILanguagePickerStrings;
	[key: string]: any;
};
