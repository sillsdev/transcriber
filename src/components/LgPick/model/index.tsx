export * from "../langPicker/types";
export * from "../localization/model";
