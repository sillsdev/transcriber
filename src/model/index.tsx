export * from './organization';
export * from './organizationMembership';
export * from './group';
export * from './groupMembership';
export * from './integration';
export * from './invitation';
export * from './mediaFile';
export * from './project';
export * from './projectType';
export * from './plan';
export * from './planType';
export * from './section';
export * from './passage';
export * from './passageSection';
export * from './user';
export * from './role';
export * from './userPassage';
export * from './activityState';
export * from './state';
export * from '../store/book/types';
export * from '../store/localization/types';
export * from '../store/localization/model';
export * from '../store/orbit/types';
export * from '../store/upload/types';
export * from '../store/context/types';
export * from '../store/langPicker/types';
export * from '../store/media/types';
export * from '../store/paratext/types';
