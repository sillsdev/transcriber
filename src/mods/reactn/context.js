Object.defineProperty(exports, '__esModule', { value: true });
var react_1 = require('react');
var default_global_state_manager_1 = require('./default-global-state-manager');
var getContext = function () {
  if (typeof react_1.createContext === 'function') {
    return react_1.createContext(default_global_state_manager_1.default);
  }
  return null;
};
exports.default = getContext();
