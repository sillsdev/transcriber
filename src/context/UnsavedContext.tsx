import React, { useState } from 'react';
// see: https://upmostly.com/tutorials/how-to-use-the-usecontext-hook-in-react
import { useGlobal } from 'reactn';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { IState, IMainStrings } from '../model';
import localStrings from '../selector/localize';
import { useRemoteSave } from '../utils';
import { useSnackBar } from '../hoc/SnackBar';
import Confirm from '../components/AlertDialog';

interface IStateProps {
  t: IMainStrings;
}
const mapStateToProps = (state: IState): IStateProps => ({
  t: localStrings(state, { layout: 'main' }),
});

interface IDispatchProps {}
const mapDispatchToProps = (dispatch: any): IDispatchProps => ({
  ...bindActionCreators({}, dispatch),
});

export interface IRowData {}

const initState = {
  checkSavedFn: (method: () => void) => {
    return;
  },
  t: {} as IMainStrings,
  handleSaveConfirmed: () => {},
  handleSaveRefused: () => {},
};

export type ICtxState = typeof initState;

interface IContext {
  state: ICtxState;
  setState: React.Dispatch<React.SetStateAction<ICtxState>>;
}

const UnsavedContext = React.createContext({} as IContext);

interface IProps extends IStateProps, IDispatchProps {
  children: React.ReactElement;
}

const UnsavedProvider = connect(
  mapStateToProps,
  mapDispatchToProps
)((props: IProps) => {
  const { t } = props;
  // const [memory] = useGlobal('memory');
  const [importexportBusy] = useGlobal('importexportBusy');
  const [busy] = useGlobal('remoteBusy');
  const [changed, setChanged] = useGlobal('changed');
  const [doSave] = useGlobal('doSave');
  const [alertOpen, setAlertOpen] = useGlobal('alertOpen'); //global because planSheet checks it
  const saveConfirm = React.useRef<() => any>();
  const { showMessage } = useSnackBar();
  const [startSave, , waitForSave] = useRemoteSave();
  const [state, setState] = useState({
    ...initState,
    t,
  });

  const checkSavedFn = (method: () => any) => {
    if (busy || importexportBusy) {
      showMessage(t.loadingTable);
      return;
    }
    if (doSave) {
      showMessage(t.saving);
      return;
    }
    if (changed) {
      saveConfirm.current = method;
      setAlertOpen(true);
    } else {
      method();
    }
  };

  const handleSaveRefused = () => {
    if (saveConfirm.current) saveConfirm.current();
    saveConfirm.current = undefined;
    setAlertOpen(false);
    setChanged(false);
  };

  const finishConfirmed = (
    savedMethod: undefined | (() => any),
    waitCount: number
  ) => {
    waitForSave(savedMethod, waitCount).catch((err) => {
      showMessage(err.message);
    });
  };

  const handleSaveConfirmed = () => {
    const savedMethod = saveConfirm.current;
    saveConfirm.current = undefined;
    showMessage(t.saving);
    startSave();
    setAlertOpen(false);
    finishConfirmed(savedMethod, 8);
  };

  return (
    <UnsavedContext.Provider
      value={{
        state: {
          ...state,
          checkSavedFn,
          handleSaveRefused,
          handleSaveConfirmed,
        },
        setState,
      }}
    >
      {props.children}
      {alertOpen && (
        <Confirm
          title={t.UnsavedData}
          text={t.saveFirst}
          yesResponse={handleSaveConfirmed}
          noResponse={handleSaveRefused}
        />
      )}
    </UnsavedContext.Provider>
  );
});
export { UnsavedContext, UnsavedProvider };
