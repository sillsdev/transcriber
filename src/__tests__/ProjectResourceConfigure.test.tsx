/* eslint-disable testing-library/no-node-access */
import { render } from '@testing-library/react';
import ProjectResourceConfigure from '../components/PassageDetail/Internalization/ProjectResourceConfigure';
import { MediaFile, MediaFileD, SectionResource } from '../model';
import { UnsavedProvider } from '../context/UnsavedContext';
import { PropsWithChildren } from 'react';
import { memory } from '../schema';
import { InitializedRecord } from '@orbit/records';

var mockMemory = memory;
var mockMediafile: MediaFile[] = [];
var mockSectionResource: SectionResource[] = [];

jest.mock('../components/passageDetail/PassageDetailPlayer', () => ({
  PassageDetailPlayer: () => <div>PassageDetailPlayer</div>,
}));
jest.mock(
  '../components/PassageDetail/Internalization/useProjectResourceSave',
  () => ({
    useProjectResourceSave: jest.fn(),
  })
);
jest.mock(
  '../components/PassageDetail/Internalization/useProjectSegmentSave',
  () => ({
    useProjectSegmentSave: jest.fn(),
  })
);
jest.mock('../control', () => ({
  ActionRow: ({ children }: PropsWithChildren) => (
    <div>
      <span>ActionRow</span>
      {children}
    </div>
  ),
  AltButton: ({ children }: PropsWithChildren) => (
    <div>
      <span>AltButton</span>
      {children}
    </div>
  ),
  GrowingSpacer: () => <div>GrowingSpacer</div>,
  LightTooltip: ({ children }: PropsWithChildren) => (
    <div>
      <span>LightTooltip</span>
      {children}
    </div>
  ),
  PriButton: ({ children }: PropsWithChildren) => (
    <div>
      <span>PriButton</span>
      {children}
    </div>
  ),
}));
jest.mock('../hoc/useOrbitData', () => ({
  useOrbitData: (recType: string) =>
    recType === 'mediafile'
      ? mockMediafile
      : recType === 'sectionresource'
      ? mockSectionResource
      : [],
}));
jest.mock('reactn', () => ({
  useGlobal: (arg: string) =>
    arg === 'memory' ? [mockMemory, jest.fn()] : [{}, jest.fn()],
}));
jest.mock('react-redux', () => ({
  useSelector: () => ({
    availableOnClipboard: 'Available on Clipboard',
    cancel: 'Cancel',
    canceling: 'Canceling',
    cantCopy: "Can't Copy",
    clipboard: 'Clipboard',
    copyToClipboard: 'Copy to Clipboard',
    createResources: 'Create Resources',
    description: 'Description',
    noData: 'No Data {0}',
    pasteError: 'Paste Errors {0}, Updated {1}',
    pasteFormat: 'Paste Format',
    pasteNoChange: 'Paste No Change',
    projectResourceConfigure: 'Project Resource Configure',
    reference: 'Reference',
    startStop: 'Start/Stop',
    suffix: 'Suffix',
    suffixTip: 'Suffix Tip',
    unusedSegment: 'Unused Segment',
  }),
  shallowEqual: jest.fn(),
}));

const defMedia: MediaFileD = {
  type: 'mediafile',
  attributes: {
    versionNumber: 1,
    eafUrl: 'General Resource',
    audioUrl: 'media/OT_Book_32___01_Jonah_______ESV_lLuke.ogg',
    s3file: '',
    duration: 0,
    contentType: '',
    audioQuality: '',
    textQuality: '',
    transcription: '',
    originalFile: 'OT_Book_32___01_Jonah_______ESV_lLuke.ogg',
    filesize: 1011383,
    position: 0,
    segments: '{}', // value null incompatible with string
    languagebcp47: '',
    link: false,
    readyToShare: false,
    performedBy: '',
    sourceSegments: '{}', // value null incompatible with string
    sourceMediaOfflineId: '',
    transcriptionstate: '',
    topic: '',
    lastModifiedBy: -1, // Add the missing property
    resourcePassageId: -1, // Add the missing property
    offlineId: '', // Add the missing property
    dateCreated: '2024-05-08T15:38:09.121Z',
    dateUpdated: '2024-05-08T15:38:09.121Z',
  },
  id: 'm1',
  relationships: {
    lastModifiedByUser: {
      data: { type: 'user', id: 'u1' },
    },
    plan: {
      data: { type: 'plan', id: 'pl1' },
    },
    artifactType: {
      data: { type: 'artifacttype', id: 'at1' },
    },
    recordedbyUser: {
      data: { type: 'user', id: 'u1' },
    },
  },
} as MediaFileD;

const runTest = (props: any) =>
  render(
    <UnsavedProvider>
      <ProjectResourceConfigure {...props} />
    </UnsavedProvider>
  );

const defItems = [
  { type: 'passage', id: 'pa1' },
  { type: 'section', id: 'se2' },
  { type: 'passage', id: 'pa2' },
  { type: 'passage', id: 'pa5' },
  { type: 'passage', id: 'pa3' },
  { type: 'passage', id: 'pa4' },
  { type: 'passage', id: 'pa6' },
];

const defPass = {
  type: 'passage',
  attributes: {
    book: 'LUK',
    title: '',
    state: 'noMedia',
    dateCreated: '2024-05-08T15:37:36.284Z',
    dateUpdated: '2024-05-08T15:37:36.284Z',
  },
  relationships: {
    lastModifiedByUser: {
      data: { type: 'user', id: 'u1' },
    },
  },
};

const passages = [
  {
    ...defPass,
    attributes: { ...defPass.attributes, sequencenum: 1, reference: '1:1-4' },
    id: 'pa1',
    relationships: {
      ...defPass.relationships,
      section: {
        data: { type: 'section', id: 'se1' },
      },
    },
  },
  {
    ...defPass,
    attributes: {
      ...defPass.attributes,
      sequencenum: 1,
      reference: '1:5-7',
      startChapter: 1,
      startVerse: 5,
      endChapter: 1,
      endVerse: 7,
    },
    id: 'pa2',
    relationships: {
      ...defPass.relationships,
      section: {
        data: { type: 'section', id: 'se2' },
      },
    },
  },
  {
    ...defPass,
    attributes: { ...defPass.attributes, sequencenum: 3, reference: '1:11-17' },
    id: 'pa3',
    relationships: {
      ...defPass.relationships,
      section: {
        data: { type: 'section', id: 'se2' },
      },
    },
  },
  {
    ...defPass,
    attributes: { ...defPass.attributes, sequencenum: 4, reference: '1:18-20' },
    id: 'pa4',
    relationships: {
      ...defPass.relationships,
      section: {
        data: { type: 'section', id: 'se2' },
      },
    },
  },
  {
    ...defPass,
    attributes: { ...defPass.attributes, sequencenum: 2, reference: '1:8-10' },
    id: 'pa5',
    relationships: {
      ...defPass.relationships,
      section: {
        data: { type: 'section', id: 'se2' },
      },
    },
  },
  {
    ...defPass,
    attributes: {
      ...defPass.attributes,
      sequencenum: 5,
      reference: '1:21-25',
      startChapter: 1,
      startVerse: 21,
      endChapter: 1,
      endVerse: 25,
    },
    id: 'pa6',
    relationships: {
      ...defPass.relationships,
      section: {
        data: { type: 'section', id: 'se2' },
      },
    },
  },
] as InitializedRecord[];

const defSec = {
  type: 'section',
  attributes: {
    state: '',
    level: 3,
    published: false,
    publishTo: '{}',
    dateCreated: '2024-05-08T15:37:36.186Z',
    dateUpdated: '2024-05-08T15:37:36.186Z',
  },
  relationships: {
    lastModifiedByUser: {
      data: { type: 'user', id: 'u1' },
    },
    plan: {
      data: { type: 'plan', id: 'pl1' },
    },
  },
};

const sections = [
  {
    ...defSec,
    attributes: {
      ...defSec.attributes,
      sequencenum: 1,
      name: 'Luke wrote this book about Jesus for Theophilus',
    },
    id: 'se1',
    relationships: {
      ...defSec.relationships,
      passages: {
        data: [{ type: 'passage', id: 'pa1' }],
      },
    },
  },
  {
    ...defSec,
    attributes: {
      ...defSec.attributes,
      sequencenum: 2,
      name: 'An angel said that John the Baptizer would be born',
    },
    id: 'se2',
    relationships: {
      ...defSec.relationships,
      passages: {
        data: [
          { type: 'passage', id: 'pa2' },
          { type: 'passage', id: 'pa5' },
          { type: 'passage', id: 'pa3' },
          { type: 'passage', id: 'pa4' },
          { type: 'passage', id: 'pa6' },
        ],
      },
    },
  },
] as InitializedRecord[];

describe('ProjectResourceConfigure', () => {
  it('should render correctly', () => {
    const props = {
      media: undefined,
      items: [],
    };
    const { container } = runTest(props);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with media', () => {
    const props = {
      media: defMedia,
      items: [],
    };
    const { container } = runTest(props);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with items', async () => {
    const props = {
      media: defMedia,
      items: defItems,
      bookData: [{ code: 'LUK', abbr: 'Lk' }],
    };
    const recs = passages.concat(sections);
    for (let rec of recs) {
      await memory.update((t) => t.addRecord(rec));
    }
    const { container } = runTest(props);
    expect(container.firstChild).toMatchSnapshot();
  });
});
