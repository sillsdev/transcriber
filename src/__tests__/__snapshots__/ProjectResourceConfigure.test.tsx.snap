// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectResourceConfigure should render correctly 1`] = `
<div
  class="MuiBox-root css-0"
>
  <div>
    PassageDetailPlayer
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-ar2ja2-MuiPaper-root"
    id="proj-res-sheet"
    style="max-height: 360px;"
  >
    <div
      class="css-106me0j"
      id="proj-res-sheet"
    >
      <div
        class="MuiStack-root css-niqf4j-MuiStack-root"
      >
        <span
          class="data-grid-container"
          tabindex="0"
        >
          <table
            class="data-grid"
          >
            <tbody />
          </table>
        </span>
        <div>
          <span>
            LightTooltip
          </span>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r0:"
              id=":r0:-label"
            >
              Suffix
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r0:"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Suffix
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <span>
      ActionRow
    </span>
    <div>
      <span>
        AltButton
      </span>
      Copy to Clipboard
    </div>
    <div>
      GrowingSpacer
    </div>
    <div>
      <span>
        PriButton
      </span>
      Create Resources
    </div>
    <div>
      <span>
        AltButton
      </span>
      Cancel
    </div>
  </div>
</div>
`;

exports[`ProjectResourceConfigure should render correctly with items 1`] = `
<div
  class="MuiBox-root css-0"
>
  <div>
    PassageDetailPlayer
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-ar2ja2-MuiPaper-root"
    id="proj-res-sheet"
    style="max-height: 360px;"
  >
    <div
      class="css-106me0j"
      id="proj-res-sheet"
    >
      <div
        class="MuiStack-root css-niqf4j-MuiStack-root"
      >
        <span
          class="data-grid-container"
          tabindex="0"
        >
          <table
            class="data-grid"
          >
            <tbody>
              <tr>
                <td
                  class="cTitle cell read-only"
                  style="width: 150px;"
                >
                  <span
                    class="value-viewer"
                  >
                    Start/Stop
                  </span>
                </td>
                <td
                  class="cTitle cell read-only"
                  style="width: 200px;"
                >
                  <span
                    class="value-viewer"
                  >
                    Reference
                  </span>
                </td>
                <td
                  class="cTitle cell read-only"
                  style="width: 300px;"
                >
                  <span
                    class="value-viewer"
                  >
                    Description
                  </span>
                </td>
              </tr>
              <tr>
                <td
                  class="lim cell"
                  style="width: 150px;"
                >
                  <span
                    class="value-viewer"
                  />
                </td>
                <td
                  class="ref cell read-only"
                  style="width: 200px;"
                >
                  <span
                    class="value-viewer"
                  >
                      1.1  Lk 1:1-4
                  </span>
                </td>
                <td
                  class="des cell"
                  style="width: 300px;"
                >
                  <span
                    class="value-viewer"
                  />
                </td>
              </tr>
              <tr>
                <td
                  class="lim cell"
                  style="width: 150px;"
                >
                  <span
                    class="value-viewer"
                  />
                </td>
                <td
                  class="ref cell read-only"
                  style="width: 200px;"
                >
                  <span
                    class="value-viewer"
                  >
                      2  An angel said that John the Baptizer would be born
                  </span>
                </td>
                <td
                  class="des cell"
                  style="width: 300px;"
                >
                  <span
                    class="value-viewer"
                  />
                </td>
              </tr>
              <tr>
                <td
                  class="lim cell"
                  style="width: 150px;"
                >
                  <span
                    class="value-viewer"
                  />
                </td>
                <td
                  class="ref cell read-only"
                  style="width: 200px;"
                >
                  <span
                    class="value-viewer"
                  >
                      2.1  Lk 1:5-7
                  </span>
                </td>
                <td
                  class="des cell"
                  style="width: 300px;"
                >
                  <span
                    class="value-viewer"
                  />
                </td>
              </tr>
              <tr>
                <td
                  class="lim cell"
                  style="width: 150px;"
                >
                  <span
                    class="value-viewer"
                  />
                </td>
                <td
                  class="ref cell read-only"
                  style="width: 200px;"
                >
                  <span
                    class="value-viewer"
                  >
                      2.2  Lk 1:8-10
                  </span>
                </td>
                <td
                  class="des cell"
                  style="width: 300px;"
                >
                  <span
                    class="value-viewer"
                  />
                </td>
              </tr>
              <tr>
                <td
                  class="lim cell"
                  style="width: 150px;"
                >
                  <span
                    class="value-viewer"
                  />
                </td>
                <td
                  class="ref cell read-only"
                  style="width: 200px;"
                >
                  <span
                    class="value-viewer"
                  >
                      2.3  Lk 1:11-17
                  </span>
                </td>
                <td
                  class="des cell"
                  style="width: 300px;"
                >
                  <span
                    class="value-viewer"
                  />
                </td>
              </tr>
              <tr>
                <td
                  class="lim cell"
                  style="width: 150px;"
                >
                  <span
                    class="value-viewer"
                  />
                </td>
                <td
                  class="ref cell read-only"
                  style="width: 200px;"
                >
                  <span
                    class="value-viewer"
                  >
                      2.4  Lk 1:18-20
                  </span>
                </td>
                <td
                  class="des cell"
                  style="width: 300px;"
                >
                  <span
                    class="value-viewer"
                  />
                </td>
              </tr>
              <tr>
                <td
                  class="lim cell"
                  style="width: 150px;"
                >
                  <span
                    class="value-viewer"
                  />
                </td>
                <td
                  class="ref cell read-only"
                  style="width: 200px;"
                >
                  <span
                    class="value-viewer"
                  >
                      2.5  Lk 1:21-25
                  </span>
                </td>
                <td
                  class="des cell"
                  style="width: 300px;"
                >
                  <span
                    class="value-viewer"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </span>
        <div>
          <span>
            LightTooltip
          </span>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r4:"
              id=":r4:-label"
            >
              Suffix
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r4:"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Suffix
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <span>
      ActionRow
    </span>
    <div>
      <span>
        AltButton
      </span>
      Copy to Clipboard
    </div>
    <div>
      GrowingSpacer
    </div>
    <div>
      <span>
        PriButton
      </span>
      Create Resources
    </div>
    <div>
      <span>
        AltButton
      </span>
      Cancel
    </div>
  </div>
</div>
`;

exports[`ProjectResourceConfigure should render correctly with media 1`] = `
<div
  class="MuiBox-root css-0"
>
  <div>
    PassageDetailPlayer
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-ar2ja2-MuiPaper-root"
    id="proj-res-sheet"
    style="max-height: 360px;"
  >
    <div
      class="css-106me0j"
      id="proj-res-sheet"
    >
      <div
        class="MuiStack-root css-niqf4j-MuiStack-root"
      >
        <span
          class="data-grid-container"
          tabindex="0"
        >
          <table
            class="data-grid"
          >
            <tbody />
          </table>
        </span>
        <div>
          <span>
            LightTooltip
          </span>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r2:"
              id=":r2:-label"
            >
              Suffix
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r2:"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Suffix
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <span>
      ActionRow
    </span>
    <div>
      <span>
        AltButton
      </span>
      Copy to Clipboard
    </div>
    <div>
      GrowingSpacer
    </div>
    <div>
      <span>
        PriButton
      </span>
      Create Resources
    </div>
    <div>
      <span>
        AltButton
      </span>
      Cancel
    </div>
  </div>
</div>
`;
